hera_cal/io.py

```diff
@@ -758,43 +759,43 @@ def get_file_lst_range(filepaths, filetype='uvh5', add_int_buffer=False):
         if filetype == 'miriad':
             uv = aipy.miriad.UV(f)
             # get integration time
-            int_time = uv['inttime'] * 2 * np.pi / (units.si.sday.in_units(units.si.s))
-            # get start and stop
-            start = uv['lst']
-            stop = start + (uv['ntimes'] - 1) * int_time
-            # add half an integration to get center of integration
-            start += int_time / 2
-            stop += int_time / 2
+            int_time = uv['inttime'] / (units.si.day.in_units(units.si.s))
+            int_time_rad = uv['inttime'] * 2 * np.pi / (units.si.sday.in_units(units.si.s))
+            # get start and stop, add half an integration
+            start_lst = uv['lst'] + int_time_rad / 2.0
+            start_time = uv['time'] + int_time / 2.0
+            # form time arrays
+            lst_array = (start_lst + np.arange(uv['ntimes']) * int_time_rad) % (2 * np.pi)
+            time_array = start_time + np.arange(uv['ntimes']) * int_time
+
         elif filetype == 'uvh5':
-            # get lsts directly from uhv5 file's header, much faster than using empty HERAData object
-            lst_array = np.array(h5py.File(f, mode='r')[u'Header'][u'lst_array'])
+            # get times directly from uvh5 file's header: faster than loading entire file via HERAData
+            with h5py.File(f, mode='r') as _f:
+                lst_array = np.ravel(_f[u'Header'][u'lst_array'])
+                time_array = np.unique(_f[u'Header'][u'time_array'])
             lst_indices = np.unique(lst_array.ravel(), return_index=True)[1]
             # resort by their appearance in lst_array, then unwrap
-            lsts = np.unwrap(lst_array.ravel()[np.sort(lst_indices)])
-            start, stop, int_time = lsts[0], lsts[-1], np.median(np.diff(lsts))
-
-        # add integration buffer to end of file if desired
-        if add_int_buffer:
-            stop += int_time
-
-        file_starts.append(start)
-        file_stops.append(stop)
-        int_times.append(int_time)
+            lst_array = np.unwrap(lst_array.ravel()[np.sort(lst_indices)])
+            int_time_rad = np.median(np.diff(lst_array))
+            int_time = np.median(np.diff(time_array))
 
-    file_starts = np.array(file_starts)
-    file_stops = np.array(file_stops)
-    int_times = np.array(int_times)
+        dlsts.append(int_time_rad)
+        dtimes.append(int_time)
+        file_lst_arrays.append(lst_array)
+        file_time_arrays.append(time_array)
 
-    # make sure times don't wrap
-    file_starts[np.where(file_starts < 0)] += 2 * np.pi
-    file_stops[np.where(file_stops >= 2 * np.pi)] -= 2 * np.pi
+    dlsts = np.asarray(dlsts)
+    dtimes = np.asarray(dtimes)
+    file_lst_arrays = np.asarray(file_lst_arrays)
+    file_time_arrays = np.asarray(file_time_arrays)
 
     if _array is False:
-        file_starts = file_starts[0]
-        file_stops = file_stops[0]
-        int_times = int_times[0]
+        dlsts = dlsts[0]
```

Comment:
Why not just `return dlsts[0], dtimes[0], file_lst_arrays[0], file_time_arrays[0]`?